refactor(multiaccount)_: remove multiaccount management code

saledjenic opened the pull request:
Deleted the entire multiaccount management implementation from the mobile package. Complexity reduced by removing unused code.

----
Reference to other accounts-management refactor PRs:
- https://github.com/status-im/status-go/pull/6605
- https://github.com/status-im/status-go/pull/6612
- https://github.com/status-im/status-go/pull/6640
- https://github.com/status-im/status-go/pull/6660
- https://github.com/status-im/status-go/pull/6668
- https://github.com/status-im/status-go/pull/6673
- https://github.com/status-im/status-go/pull/6686
- https://github.com/status-im/status-go/pull/6687
- https://github.com/status-im/status-go/pull/6692

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (37)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 3f456554 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6660/1/) | 2025-06-11 09:55:01 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250611-095157-3f4565-pr6660.aar) |
| :heavy_check_mark: | 3f456554 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6660/1/) | 2025-06-11 09:55:15 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250611-095202-3f4565-pr6660.zip) |
| :heavy_check_mark: | 3f456554 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6660/1/) | 2025-06-11 09:55:27 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250611-095158-3f4565-pr6660.zip) |
| :heavy_check_mark: | 3f456554 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6660/1/) | 2025-06-11 09:56:53 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250611-095209-3f4565-pr6660.zip) |
| :heavy_check_mark: | 3f456554 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6660/1/) | 2025-06-11 09:57:35 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250611-095204-3f4565-pr6660.zip) |
| :heavy_check_mark: | 3f456554 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6660/1/) | 2025-06-11 09:58:19 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250611-095158-3f4565-pr6660.zip) |
| :heavy_check_mark: | 3f456554 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6660/1/) | 2025-06-11 10:03:21 | ~11 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250611-095200-3f4565-pr6660.zip) |
| :heavy_multiplication_x: | 3f456554 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/1/) | 2025-06-11 10:04:35 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/1//consoleText) |
| :heavy_check_mark: | 3f456554 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/1/) | 2025-06-11 10:18:54 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 628cbad9 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6660/2/) | 2025-06-11 11:53:06 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250611-114956-628cba-pr6660.zip) |
| :heavy_check_mark: | 628cbad9 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6660/2/) | 2025-06-11 11:53:13 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250611-114949-628cba-pr6660.aar) |
| :heavy_check_mark: | 628cbad9 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6660/2/) | 2025-06-11 11:53:41 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250611-114951-628cba-pr6660.zip) |
| :heavy_check_mark: | 628cbad9 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6660/2/) | 2025-06-11 11:54:36 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250611-115001-628cba-pr6660.zip) |
| :heavy_check_mark: | 628cbad9 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6660/2/) | 2025-06-11 11:55:23 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250611-114958-628cba-pr6660.zip) |
| :heavy_check_mark: | 628cbad9 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6660/2/) | 2025-06-11 11:56:58 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250611-114950-628cba-pr6660.zip) |
| :heavy_check_mark: | 628cbad9 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6660/2/) | 2025-06-11 12:01:16 | ~11 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250611-114953-628cba-pr6660.zip) |
| :heavy_multiplication_x: | 628cbad9 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/2/) | 2025-06-11 12:05:15 | ~15 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/2//consoleText) |
| :heavy_check_mark: | 628cbad9 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/2/) | 2025-06-11 12:15:41 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/2//consoleText) |
| :heavy_check_mark: | 628cbad9 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/3/) | 2025-06-12 14:17:19 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 073781a5 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6660/3/) | 2025-06-12 19:11:56 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250612-190921-073781-pr6660.aar) |
| :heavy_check_mark: | 073781a5 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6660/3/) | 2025-06-12 19:12:32 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-190924-073781-pr6660.zip) |
| :heavy_check_mark: | 073781a5 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6660/3/) | 2025-06-12 19:13:27 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250612-190929-073781-pr6660.zip) |
| :heavy_check_mark: | 073781a5 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6660/3/) | 2025-06-12 19:13:47 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250612-190926-073781-pr6660.zip) |
| :heavy_check_mark: | 073781a5 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6660/3/) | 2025-06-12 19:14:19 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250612-190933-073781-pr6660.zip) |
| :heavy_check_mark: | 073781a5 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6660/3/) | 2025-06-12 19:16:19 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250612-190922-073781-pr6660.zip) |
| :heavy_check_mark: | 073781a5 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6660/3/) | 2025-06-12 19:19:23 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250612-190924-073781-pr6660.zip) |
| :heavy_check_mark: | 073781a5 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/4/) | 2025-06-12 19:22:20 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/4//consoleText) |
| :heavy_check_mark: | 073781a5 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/3/) | 2025-06-12 19:35:28 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 7182537e | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6660/4/) | 2025-06-16 10:43:57 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-104054-718253-pr6660.zip) |
| :heavy_check_mark: | 7182537e | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6660/4/) | 2025-06-16 10:44:20 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250616-104049-718253-pr6660.aar) |
| :heavy_check_mark: | 7182537e | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6660/4/) | 2025-06-16 10:45:14 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-104058-718253-pr6660.zip) |
| :heavy_check_mark: | 7182537e | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6660/4/) | 2025-06-16 10:45:37 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250616-104050-718253-pr6660.zip) |
| :heavy_check_mark: | 7182537e | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6660/4/) | 2025-06-16 10:45:50 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250616-104101-718253-pr6660.zip) |
| :heavy_check_mark: | 7182537e | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6660/4/) | 2025-06-16 10:47:30 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-104050-718253-pr6660.zip) |
| :heavy_check_mark: | 7182537e | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6660/4/) | 2025-06-16 10:55:24 | ~14 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-104051-718253-pr6660.zip) |
| :heavy_multiplication_x: | 7182537e | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/5/) | 2025-06-16 11:05:32 | ~24 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/5//consoleText) |
| :heavy_check_mark: | 7182537e | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/4/) | 2025-06-16 11:15:29 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/4//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | de287afd | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6660/5/) | 2025-06-16 11:42:39 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250616-114008-de287a-pr6660.aar) |
| :heavy_check_mark: | de287afd | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6660/5/) | 2025-06-16 11:43:15 | ~2 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-114015-de287a-pr6660.zip) |
| :heavy_check_mark: | de287afd | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6660/5/) | 2025-06-16 11:43:44 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-114012-de287a-pr6660.zip) |
| :heavy_check_mark: | de287afd | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6660/5/) | 2025-06-16 11:44:57 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250616-114009-de287a-pr6660.zip) |
| :heavy_check_mark: | de287afd | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6660/5/) | 2025-06-16 11:45:39 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-114017-de287a-pr6660.zip) |
| :heavy_check_mark: | de287afd | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6660/5/) | 2025-06-16 11:47:11 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250616-114021-de287a-pr6660.zip) |
| :heavy_check_mark: | de287afd | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/6/) | 2025-06-16 11:50:29 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6660/6//consoleText) |
| :heavy_check_mark: | de287afd | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6660/5/) | 2025-06-16 11:52:31 | ~12 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-114013-de287a-pr6660.zip) |
| :heavy_check_mark: | de287afd | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/5/) | 2025-06-16 12:08:41 | ~28 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6660/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3f83496c | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6660/6/) | 2025-06-17 12:17:34 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-121353-3f8349-pr6660.zip) |
| :heavy_check_mark: | 3f83496c | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6660/6/) | 2025-06-17 12:20:26 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-121357-3f8349-pr6660.zip) |
| :heavy_check_mark: | 3f83496c | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6660/6/) | 2025-06-17 12:21:20 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250617-121348-3f8349-pr6660.zip) |
| :heavy_check_mark: | 3f83496c | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6660/6/) | 2025-06-17 12:21:36 | ~7 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250617-121402-3f8349-pr6660.zip) |

igor-sirotin commented on mobile/accounts.go:
I'd leave it in the previous filename for clearer git history

igor-sirotin reviewed (approved):
Did I get it right that with this PR we're breaking compatibility with `status-mobile`?
I guess it's fine, just a bit scary 😅

saledjenic committed:
refactor(multiaccount)_: remove multiaccount management code

Deleted the entire multiaccount management implementation from the mobile package.
Complexity reduced by removing unused code.

saledjenic committed:
refactor(account-generator)_: simplify account management and enhance functionality

- Removed the errors.go file, consolidating error handling within the generator package.
- Refactored account creation methods to improve clarity and usability, introducing CreateAccountFromMnemonic, CreateAccountFromPrivateKey, and CreateAccountFromKey functions.
- Updated tests to align with the new account structure and ensure coverage.
- Enhanced README documentation to reflect the new account creation and derivation methods.

saledjenic committed:
refactor(account-management)_: decouple and enhance keystore functionality

- Removed the old GethManager and associated methods, moving to a new adapter-based approach for keystore management.
- Previously Introduced keystore adapter, updated, improved and moved from internal.
- Updated account management methods to utilize the new adapter, enhancing clarity and usability.
- Refactored tests to align with the new structure and ensure coverage.
- Cleaned up unused code and improved overall readability of the account package.

saledjenic committed:
refactor(account-management)_: enhance keystore integration and account handling

- Updated tests to utilize the new account management.
- Refactored account loading and creation methods to align with the new structure, enhancing maintainability.
- Cleaned up unused code and improved overall readability in the account package.
- This refactor further decouples the account management from direct dependencies on the go-ethereum keystore.

saledjenic committed:
refactor(account-management)_: remove onboarding functionality and related tests

- Deleted the onboarding-related methods and tests from the account package.
- Cleaned up unused code.
- Adjusted error handling by removing onboarding-specific errors.
